### PR TITLE
detect-engine-sigorder: Convert unittests to new FAIL/PASS API

### PR DESCRIPTION
**Pull request by hsadia538**
- [X] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Previous PR: #8125 

Link to [redmine](https://redmine.openinfosecfoundation.org/issues/4029) ticket:

Describe changes:

- Fix clang-format

### CODE REVIEW

File: src/detect-engine-sigorder.c [hunk 1]
```diff
@@ -860,8 +859,8 @@ static int SCSigOrderingTest01(void)
     DetectEngineCtxFree(de_ctx);
 
     return (i == 5);
- end:
-    return 0;
+
+    PASS;
```
Comment (jlucovsky) [inline src/detect-engine-sigorder.c:863]: This statement will never execute because of the `return` statement that precedes it

**Review by jlucovsky — COMMENTED** [review]
Looks good.

File: src/detect-engine-sigorder.c (hunk 1 above)
Comment (hsadia538) [reply src/detect-engine-sigorder.c:863]: Do you think I should add a `FAIL_IF_NOT(i==5)` to replace this

**Comment by hsadia538** [comment]
There are 13 tests in total and some of them are quite lengthy. This file is also listed [Here on redmine suri verify test list](https://redmine.openinfosecfoundation.org/issues/4911)
Should I try the conversion?

File: src/detect-engine-sigorder.c (hunk 1 above)
Comment (victorjulien) [reply src/detect-engine-sigorder.c:863]: yes, that makes sense

File: src/detect-engine-sigorder.c [hunk 2]
```diff
@@ -2082,49 +1900,36 @@ static int SCSigOrderingTest12(void)
...
-    if (p != NULL)
-        SCFree(p);
-    if (de_ctx != NULL) {
-        SigCleanSignatures(de_ctx);
-        SigGroupCleanup(de_ctx);
-        DetectEngineCtxFree(de_ctx);
-    }
+    FAIL_IF_NOT(UTHCheckPacketMatchResults(p, sids, results, 2));
+
+    SCFree(p);
```
Comment (victorjulien) [inline src/detect-engine-sigorder.c:1905]: noticed this now. Packets allocated by `UTHBuildPacket` should be freed by `PacketFree`. Can you update the tests to do this in the next PR?

**Review by victorjulien — CHANGES_REQUESTED** [review]

**Comment by victorjulien** [comment]
> There are 13 tests in total and some of them are quite lengthy. This file is also listed [Here on redmine suri verify test list](https://redmine.openinfosecfoundation.org/issues/4911) Should I try the conversion?

Good idea. Lets do that after this cleanup work.

File: src/detect-engine-sigorder.c (hunk 2 above)
Comment (hsadia538) [reply src/detect-engine-sigorder.c:1905]: I was able to find one occurance only.

**Comment by hsadia538** [comment]
Followed by #8131

File: src/detect-engine-sigorder.c (hunk 2 above)
Comment (hsadia538) [reply src/detect-engine-sigorder.c:1905]: I get the following when I try to free the memory and use it at the end right after `SCFree(p);`
```
suricata(60098,0x102af4580) malloc: *** error for object 0x140035200: pointer being freed was not allocated
suricata(60098,0x102af4580) malloc: *** set a breakpoint in malloc_error_break to debug
```

If I try to free after line 1903 then I get 
```
FLOW MEMORY IN USE 11748608
FAILED
```

**Comment by victorjulien** [comment]
Why was this reopened?

**Comment by hsadia538** [comment]
> Why was this reopened?
Opened again to gather feedback on updates.
Tests were failing again. I have added inline comments

File: src/detect-engine-sigorder.c (hunk 2 above)
Comment (victorjulien) [reply src/detect-engine-sigorder.c:1905]: I think my statement was incorrect. In other tests I see another free function being used:
```c
UTHFreePackets(&p, 1);
```
E.g. see `ModbusParserTest03`

Comment (hsadia538) [reply src/detect-engine-sigorder.c:1905]: I have tried both

**Pull request closed without merging**